### PR TITLE
Colorize edge with int data type

### PR DESCRIPTION
**Pull request by seanshpark**
This will colorize edge with int data type.
This is to better visualize quantized models.

### CODE REVIEW

**Comment by seanshpark** [comment]
sample capture of [Quantize_001.zip](https://github.com/user-attachments/files/15878340/Quantize_001.zip) tflite model


dark | light
--- | ---
image | image

File: source/grapher.css [hunk 1]
```diff
@@ -69,6 +69,10 @@
 
 .edge-path { stroke: #000; stroke-width: 1px; fill: none; marker-end: url("#arrowhead"); }
 .edge-path-hit-test { pointer-events: stroke; stroke-width: 0.5em; fill: none; stroke: #000; stroke-opacity: 0.001; }
+.edge-path-b4 { stroke: #70e; stroke-width: 1x; fill: none; marker-end: url("#arrowhead"); }
+.edge-path-b8 { stroke: #04e; stroke-width: 1px; fill: none; marker-end: url("#arrowhead"); }
+.edge-path-b16 { stroke: #080; stroke-width: 1px; fill: none; marker-end: url("#arrowhead"); }
+.edge-path-b32 { stroke: #992; stroke-width: 1px; fill: none; marker-end: url("#arrowhead"); }
```
Comment (seanshpark) [inline source/grapher.css:75]: I'm not good at colors. Please choose with your preference :)

File: source/grapher.js [hunk 2]
```diff
@@ -737,11 +737,53 @@ grapher.Edge = class {
         const createElement = (name) => {
             return document.createElementNS('http://www.w3.org/2000/svg', name);
         };
+        const getDatatype = (from) => {
+            if (from && from.value) {
+                const from_value = from.value;
+                let from_value_src;
+                if (from_value.outputs && from_value.outputs[0].value) {
```
Comment (kylesayrs) [inline source/grapher.js:744]:
```suggestion
                if (from_value.outputs && from_value.outputs[0]?.value) {
```

File: source/grapher.js [hunk 3]
```diff
@@ -737,11 +737,53 @@ grapher.Edge = class {
...
+                            return value._type._dataType;
+                        }
+                    }
+                }
+            }
+            return '';
+        };
+        const edgePathClass = (dataType) => {
+            const dt = dataType.toLowerCase();
+            if (dt === 'uint8' || dt === 'int8') {
```
Comment (kylesayrs) [inline source/grapher.js:765]: nit: may consider swapping to a switch statement here

File: source/grapher.js [hunk 4]
```diff
@@ -737,11 +737,53 @@ grapher.Edge = class {
         const createElement = (name) => {
             return document.createElementNS('http://www.w3.org/2000/svg', name);
         };
+        const getDatatype = (from) => {
+            if (from && from.value) {
```
Comment (kylesayrs) [inline source/grapher.js:741]:
```javascript
let from_value_src = from?.value?.outputs?.[0]?.value;
```

File: source/grapher.js (hunk 2 above)
Comment (lutzroeder) [reply source/grapher.js:744]: Currently not use optional chaining. Check browser support matrix and apply consistently across the codebase.

**Review by lutzroeder — CHANGES_REQUESTED** [review]
A few questions:

1. There are many more data types, why hardcode this set and how would this extend to other types?
2. Could this be overwhelming or confusing to other users?
3. What other solutions might work to solve this?

**Comment by lutzroeder** [comment]
Closing as the pull request been stale for two weeks. Please comment if there are updates and it should be reopened.

**Pull request closed without merging**